src/main/java/org/json/JSONObject.java

```diff
@@ -2295,7 +2295,7 @@ protected static Number stringToNumber(final String val) throws NumberFormatExce
     // Changes to this method must be copied to the corresponding method in
     // the XML class to keep full support for Android
     public static Object stringToValue(String string) {
```

Comment:
As this is a public method, the contract has to remain the same. This means supporting a `null` parameter value.